### PR TITLE
chore(bzlmod): add trivial pass-through module extension for oci_pull

### PR DESCRIPTION
**Pull request by alexeagle**
However it trips over the labels in our generated build files making assumptions about repository names, needs some fixing

### CODE REVIEW

File: oci/extensions.bzl [hunk 1]
```diff
@@ -1,6 +1,21 @@
...
+    "image": attr.string(doc = """the remote image without a tag, such as gcr.io/bazel-public/bazel"""),
+    "platforms": attr.string_list(doc = """for multi-architecture images, a dictionary of the platforms it supports
+            This creates a separate external repository for each platform, avoiding fetching layers."""),
+    "digest": attr.string(doc = """the digest string, starting with "sha256:", "sha512:", etc.
+            If omitted, instructions for pinning are provided."""),
+    "tag": attr.string(doc = """a tag to choose an image from the registry.
+            Exactly one of `tag` and `digest` must be set.
+            Since tags are mutable, this is not reproducible, so a warning is printed."""),
+    "reproducible": attr.bool(doc = """Set to False to silence the warning about reproducibility when using `tag`.""", default = True),
+})
```
Comment (kormide) [inline oci/extensions.bzl:18]: Can you just share the `attrs` object with the rule? Then add any extra attributes for the macro (if there is one).

Comment (kormide) [reply oci/extensions.bzl:18]: https://github.com/aspect-build/rules_js/blob/main/npm/extensions.bzl#L157

Comment (alexeagle) [reply oci/extensions.bzl:18]: I tried that at first, however this calls a wrapper macro where the argument are nearly non-overlapping. For instance the underlying rule has a `platform` attribute, but the macro has a loop so it accepts `platforms` instead.

**Comment by alexeagle** [comment]
I made an example to test this, and there's a bug. Problem here is

```
ERROR: /home/alexeagle/Projects/bazel-examples/oci_go_image/BUILD.bazel:34:10: no such target '@rules_oci~override~oci~distroless_base//:image': target 'image' not declared in package '' defined by /shared/cache/bazel/user_base/359b45a24cf55cc22380e941fe3bcf54/external/rules_oci~override~oci~distroless_base/BUILD.bazel
```

which is caused by https://github.com/bazel-contrib/rules_oci/blob/main/oci/pull.bzl#L246

I wanted to have nice syntax sugar for users to choose a base image like just `base = "@distroless_static"` but one option here is to give that up and require a label like `@distroless_static//:image` - then we don't need to know the repo name here to create a target with a matching name.

**Comment by kormide** [comment]
> I made an example to test this, and there's a bug. Problem here is
> 
> ```
> ERROR: /home/alexeagle/Projects/bazel-examples/oci_go_image/BUILD.bazel:34:10:
> no such target '@rules_oci~override~oci~distroless_base//:distroless_base': target 'distroless_base' not declared in package '' 
> defined by /shared/cache/bazel/user_base/359b45a24cf55cc22380e941fe3bcf54/external/rules_oci~override~oci~distroless_base/BUILD.bazel
> ```
> 
> which is caused by https://github.com/bazel-contrib/rules_oci/blob/main/oci/pull.bzl#L246
> 
> I wanted to have nice syntax sugar for users to choose a base image like just `base = "@distroless_static"` but one option here is to give that up and require a label like `@distroless_static//:image` - then we don't need to know the repo name here to create a target with a matching name.

I'm missing some pieces here. So you want to be able to pass in `base = "@distroless_static"` into `oci_image`? As in you want to pass a string rather than a label and have it do the equivalent of appending the `//:image` target?

**Comment by alexeagle** [comment]
okay the downstream example is green now!

**Review by kormide — APPROVED** [review]

**Pull request merged**